### ISSUE TITLE
Do not raise error "Do not use Localhost in your code" when "localhost" in comment

### ISSUE THREAD

**Issue by leoloso** [10up/plugin-check#267]
### Is your enhancement related to a problem? Please describe.

I'm given the following error:

```
Do not use Localhost in your code. Detected: http://localhost
```

However, my code doesn't use `localhost` at all. The only occurrence is inside a comment (from a 3rd-party package):

```php
/**
 * Returns the path being requested relative to the executed script.
 *
 * The path info always starts with a /.
 *
 * Suppose this request is instantiated from /mysite on localhost:
 *
 *  * http://localhost/mysite              returns an empty string
 *  * http://localhost/mysite/about        returns '/about'
 *  * http://localhost/mysite/enco%20ded   returns '/enco%20ded'
 *  * http://localhost/mysite/about?var=1  returns '/about'
 *
 * @return string The raw path (i.e. not urldecoded)
 */
public function getPathInfo(): string
{
    return $this->pathInfo ??= $this->preparePathInfo();
}
```

I guess the error should not be raised when the match happens inside a comment.

### Designs

<img width="576" alt="Screenshot 2023-09-22 at 12 01 33 PM" src="https://github.com/10up/plugin-check/assets/1981996/a9066ccf-2248-4b01-96ac-45d9e17c9d1f">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
